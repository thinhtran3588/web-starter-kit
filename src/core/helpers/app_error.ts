export class AppError extends Error {
  code: string;

  constructor(code: string, message: string) {
    super(message);
    this.code = code;
    // Set the prototype explicitly.
    Object.setPrototypeOf(this, AppError.prototype);
  }
}
