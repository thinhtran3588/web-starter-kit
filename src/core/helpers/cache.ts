// eslint-disable-next-line @typescript-eslint/no-explicit-any
const cacheStorage: Record<string, any> = {};

const get = <T>(key: string): T => {
  return cacheStorage[key];
};

const clear = (key: string): void => {
  delete cacheStorage[key];
};

const expire = (key: string, time: number): void => {
  setTimeout(() => {
    clear(key);
  }, time);
};

const set = <T>(key: string, value: T, expireTime?: number): void => {
  cacheStorage[key] = value;
  if (expireTime !== undefined) {
    expire(key, expireTime);
  }
};

export const cache = {
  get,
  set,
  clear,
  expire,
};
