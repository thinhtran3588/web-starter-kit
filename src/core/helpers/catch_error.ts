/* eslint-disable @typescript-eslint/no-explicit-any */
import { handleError } from './handle_error';

export const catchError = <T>(
  action: T,
  setIsBusy?: (f: (draft: boolean) => boolean | void) => void,
  matchedCodes?: { [code: string]: string },
  ignoreCodes?: { [code: string]: true },
): T => {
  return ((async (...params: any[]): Promise<any> => {
    try {
      setIsBusy && setIsBusy(() => true);
      return await (action as any)(...(params as any));
    } catch (error) {
      handleError(error, matchedCodes, ignoreCodes);
    } finally {
      setIsBusy && setIsBusy(() => false);
    }
    return undefined;
  }) as unknown) as T;
};
