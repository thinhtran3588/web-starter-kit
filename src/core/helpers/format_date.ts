import dayjs from 'dayjs';
import { config } from '@app/config';
import { FieldValueType } from '../interfaces';

export const formatDate = (value: FieldValueType): string => {
  if (!value) {
    return '';
  }
  return dayjs(value).format(config.dateFormat);
};
