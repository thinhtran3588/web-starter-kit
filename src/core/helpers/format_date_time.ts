import dayjs from 'dayjs';
import { config } from '@app/config';
import { FieldValueType } from '../interfaces';

export const formatDateTime = (value: FieldValueType): string => {
  if (!value) {
    return '';
  }
  return dayjs.unix(parseFloat((value as string).substring(0, 10))).format(config.dateTimeFormat);
};
