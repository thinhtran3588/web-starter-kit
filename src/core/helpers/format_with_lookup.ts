import { FieldValueType, PickerDataItem } from '../interfaces';

export const formatWithLookup = (lookup: PickerDataItem<string>[]) => (value: FieldValueType) => {
  if (!value) {
    return '';
  }
  const loginType = lookup.find((m) => m.value === value);
  return loginType ? loginType.label : '';
};
