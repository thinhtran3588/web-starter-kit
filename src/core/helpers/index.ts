export * from './format_with_lookup';
export * from './format_date';
export * from './format_date_time';
export * from './show_notification';
export * from './catch_error';
export * from './handle_error';
export * from './app_error';
export * from './sanitize_form_data';
export * from './get_error_message';
export * from './sleep';
