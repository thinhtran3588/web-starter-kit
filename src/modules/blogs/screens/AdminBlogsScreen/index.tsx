import React from 'react';
import { AdminLayout } from '@app/components';
import { WithTranslation, AuthProps, withTranslation } from '@app/core';
import { withAuth } from '@app/hoc/WithAuth';
import { useStyles } from './styles';

type Props = WithTranslation & AuthProps;

export const Screen = (props: Props): JSX.Element => {
  /* --- variables & states - begin --- */
  const { t, userMenuItems } = props;
  const classes = useStyles();
  /* --- variables & states - end --- */

  return (
    <AdminLayout title={t('users')} userMenuItems={userMenuItems}>
      <h1 className={classes.header}>AdminBlogs</h1>
    </AdminLayout>
  );
};

const ScreenBeforeTranslation = withAuth(Screen, (validatePermissions) => validatePermissions('blogs', 'viewAny'));

ScreenBeforeTranslation.getInitialProps = async () => {
  return {
    namespacesRequired: ['common', 'admin_blogs'],
  };
};

export const AdminBlogsScreen = withTranslation('admin_blogs')(ScreenBeforeTranslation);
