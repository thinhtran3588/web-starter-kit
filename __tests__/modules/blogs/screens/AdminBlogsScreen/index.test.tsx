import React from 'react';
import { render } from '@testing-library/react';
import { AdminBlogsScreen } from '@app/modules/blogs/screens/AdminBlogsScreen';
import { TestBaseComponent } from '@test/helpers/TestBaseComponent';

describe('@app/modules/AdminBlogs/screens/AdminBlogs', () => {
  beforeEach(() => {});

  it('renders successfully', async () => {
    const { baseElement } = render(
      <TestBaseComponent>
        <AdminBlogsScreen />
      </TestBaseComponent>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
