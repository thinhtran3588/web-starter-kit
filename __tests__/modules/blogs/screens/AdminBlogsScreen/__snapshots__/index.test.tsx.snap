// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@app/modules/AdminBlogs/screens/AdminBlogs renders successfully 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-1"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;
